upf/pfcp.c

```diff
@@ -6003,6 +6171,48 @@ static struct pfcp_group_ie_def pfcp_update_access_forwarding_action_information
     },
   };
 
+static struct pfcp_group_ie_def pfcp_ue_ip_address_pool_group[] =
+  {
+    [UE_IP_ADDRESS_POOL_INFORMATION_POOL_IDENTIFY] = {
+      .type = PFCP_IE_UE_IP_ADDRESS_POOL_IDENTITY,
+      .offset = offsetof(pfcp_ue_ip_address_pool_information_t,
+			 ue_ip_address_pool_identity)
+    },
+    [UE_IP_ADDRESS_POOL_INFORMATION_NETWORK_INSTANCE] = {
+      .type = PFCP_IE_NETWORK_INSTANCE,
+      .offset = offsetof(pfcp_ue_ip_address_pool_information_t,
+			 network_instance)
+    },
+    [UE_IP_ADDRESS_POOL_INFORMATION_IP_VERSION] = {
+      .type = PFCP_IE_IP_VERSION,
+      .offset = offsetof(pfcp_ue_ip_address_pool_information_t,
+			 ip_version)
+    },
+    [UE_IP_ADDRESS_POOL_INFORMATION_BBF_NAT_PORT_BLOCK] = {
+      .type = PFCP_IE_BBF_NAT_PORT_BLOCK,
+      .vendor = VENDOR_BBF,
+      .is_array = true,
+      .offset = offsetof(pfcp_ue_ip_address_pool_information_t,
+                         port_blocks)
+    },
+  };
+
+#define encode_ip_version encode_u8_ie
+#define decode_ip_version decode_u8_ie
```

Comment:
similar IEs implement some minimal error checking on the flags, e.g. check that at least one bit is set.

Reply:
introduced enc/dec functions for this, thanks